test/test_fcntl.rs

```diff
@@ -584,3 +584,17 @@ fn test_f_get_path() {
         tmp.path().canonicalize().unwrap()
     );
 }
+
+#[cfg(target_os = "freebsd")]
+#[test]
+fn test_f_kinfo() {
+    use nix::fcntl::*;
+    use std::{os::unix::io::AsRawFd, path::PathBuf};
+
+    let tmp = NamedTempFile::new().unwrap();
+    let fd = tmp.as_raw_fd();
+    let mut path = PathBuf::new();
+    let res = fcntl(fd, FcntlArg::F_KINFO(&mut path)).expect("get path failed");
+    assert_ne!(res, -1);
+    assert_eq!(path.as_path(), tmp.path());
```

Comment:
We have `impl<'a> PartialEq<&'a Path> for PathBuf` so that `.as_path()` in the left is not necessary